src/arch/x64/devices/high_precision_timer.rs

```diff
@@ -28,7 +28,9 @@ pub fn rdtsc() -> u64 {
     unsafe {
         let lower: u64;
         let higher: u64;
-        asm!(" rdtsc "
+        asm!("
+            lfence
+            rdtsc"
```

Comment:
Can you maybe use `rdtscp`?

If not, I think you may need another `lfence` after the `rdtsc`.

Reply:
I wasn't sure whether to put it before or after. You can use rdtscp, but it seems not to be available on some systems.